src/urllib3/_collections.py

```diff
@@ -301,14 +311,23 @@ def discard(self, key: str) -> None:
         except KeyError:
             pass
 
-    def add(self, key: str, val: str) -> None:
+    def add(self, key: str, val: str, *, combine: Optional[bool] = None) -> None:
```

Comment:
`combine` can have a default value of `False` since that matches existing behavior. Kudos for making `combine` keyword-only! :clap: